ports/unix/modffi.c

```diff
@@ -365,6 +385,48 @@ STATIC void ffifunc_print(const mp_print_t *print, mp_obj_t self_in, mp_print_ki
     mp_printf(print, "<ffifunc %p>", self->func);
 }
 
+STATIC unsigned long long ffi_get_int_value(mp_obj_t o) {
+    if (MP_OBJ_IS_SMALL_INT(o)) {
+        return MP_OBJ_SMALL_INT_VALUE(o);
+    } else {
+        unsigned long long res;
+        mp_obj_int_to_bytes_impl(o, MP_ENDIANNESS_BIG, sizeof(res), (byte *) &res);
+        return res;
+    }
+}
+
+STATIC ffi_union_t ffi_int_obj_to_ffi_union(mp_obj_t o, const char argtype) {
+    ffi_union_t ret = {0};
+    unsigned long long val = ffi_get_int_value(o);
+    switch (argtype) {
+        case 'b':
+        case 'B':
+            ret.B = val;
+            break;
+        case 'h':
+        case 'H':
+            ret.H = val;
+            break;
+        case 'i':
+        case 'I':
+            ret.I = val;
+            break;
+        case 'l':
+        case 'L':
+            ret.L = val;
```

Comment:
I think for all these <=32 bit sizes it's simpler and more efficient to just call `mp_obj_int_get_truncated(o)` directly.  Only for Q/q is special care needed (and then only on 32-bit archs).

Reply:
I did this in #7358